network/shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/BlockTransferMessage.java

```diff
@@ -40,7 +41,8 @@
 
   /** Preceding every serialized message is its type, which allows us to deserialize it. */
   public static enum Type {
-    OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4);
+    OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4),
+    HEARTBEAT(5);
```

Comment:
this is the part that's the most worrying since it doesn't core stuff. Can you run some end-to-end tests of Spark standalone mode with external shuffle service to ensure that everything continues to work as expected?